samples/xbee/demos/EndToEndAgricultureSample/main.py

```diff
@@ -904,31 +930,101 @@ def get_next_random(value, max_value, min_value, max_delta):
     return value
 
 
-def get_rain():
+def get_temp():
+    """
+    Returns the temperature. The value is obtained from the
+    meteo station.
+
+    Returns:
+        The new temperature in ºC.
+    """
+    return temp
+
+
+def get_rain(simulation=False):
     """
     Returns the rain during the day. The value is randomly calculate based on
     the current weather condition.
 
     Returns:
         The rain in mm.
     """
-    global rain
+    rain_var = 0
+
+    if simulation:
+        # Report rain only if the condition is 'rainy' (and not always).
+        if weather_condition == CONDITION_RAINY and random.random() > 0.7:
+            rain_var += round(random.random(), 2)
+
+    else:
+        rain_var = rain
 
-    # Report rain only if the condition is 'rainy' (and not always).
-    if weather_condition == CONDITION_RAINY and random.random() > 0.7:
-        rain += round(random.random(), 2)
-    return rain
+    # Conversion from inch to mm
+    return rain_var * 25.4
 
 
-def get_wind():
+def get_wind(simulation=False):
     """
-    Returns the wind speed. The value is randomly calculated based on the
-    current wind speed value.
+    Returns the wind speed. If simulation is set to True, the value is randomly
+    calculated based on the current wind speed value.
+    If it is set to False, then the value is obtained from the meteo station.
 
     Returns:
         The new wind speed in km/h.
     """
-    return get_next_random(wind, WIND_MAX, WIND_MIN, WIND_DELTA)
+
+    if simulation:
+        return get_next_random(wind, WIND_MAX, WIND_MIN, WIND_DELTA)
+    else:
+        # Conversion from miles/h to km/h
+        return round(wind * 1.609, 1)
```

Comment:
It would be more elegant to have a separated function to convert to km/h and call it here.